keras/src/backend/config.py

```diff
@@ -403,3 +411,36 @@ def max_steps_per_epoch():
             `None`, no limit is applied.
     """
     return _MAX_STEPS_PER_EPOCH
+
+
+if not os.path.exists(_KERAS_DIR):
+    try:
+        os.makedirs(_KERAS_DIR)
+    except OSError:
+        # Except permission denied and potential race conditions
+        pass
+
+if not os.path.exists(_config_path):
+    _current_nnx_status_for_saving = is_nnx_backend_enabled()
+    _config_to_save = {
+        "floatx": floatx(),
+        "epsilon": epsilon(),
+        "backend": _BACKEND,  # Use the final _BACKEND value
+        "image_data_format": image_data_format(),
+        _NNX_ENABLED_KEY: _current_nnx_status_for_saving,
+    }
+    try:
+        with open(_config_path, "w") as f:
+            f.write(json.dumps(_config_to_save, indent=4))
+    except IOError:
+        # Except permission denied.
+        pass
+
+if "KERAS_NNX_ENABLED" in os.environ:
+    env_val = os.environ["KERAS_NNX_ENABLED"].lower()
+    if env_val == "true":
+        _initial_nnx_value_from_config = True
```

Comment:
Try to rewrite the changes to this file trying to follow the local style a little more. This is overcomplicated.

- no need to move big blocks around.
- keep a constant `_NNX_ENABLED` at the top of the file, somewhere after `_BACKEND`.
- set it from config the same way `_BACKEND` is handled
- write it to config the same way `_BACKEND` is handled

Get rid of `_NNX_ENABLED_KEY`, `_initial_nnx_value_from_config`, `_current_nnx_status_for_saving`.

Reply:
There are a lot of circular import errors, because of which i had a move some of these blocks lower.